test/Rating.js

```diff
@@ -1,9 +1,127 @@
+		assert.strictEqual('', rating.element.style.display);
+	});
+
+	it('should render specified label.', function() {
+		rating = new Rating({
+			label: "This is an awesome Metal Component"
+		}).render();
+		assert.strictEqual("This is an awesome Metal Component", rating.element.textContent);
+	});
+
+	it('should highlight until the item clicked', function(done) {
+		rating = new Rating().render();
+
+		var elements = rating.element.querySelectorAll('a');
+		dom.triggerEvent(elements[2], 'click');
+
+		setTimeout(function(){
+			elements = rating.element.querySelectorAll('a');
+			assert.strictEqual(elements[0].className, rating.cssClasses.on);
+			assert.strictEqual(elements[1].className, rating.cssClasses.on);
+			assert.strictEqual(elements[2].className, rating.cssClasses.on);
+			assert.strictEqual(elements[3].className, rating.cssClasses.off);
+			assert.strictEqual(elements[4].className, rating.cssClasses.off);
+			done();
+		}, 25);
+	});
+
+	it('should set the rate value by click', function(done) {
+		rating = new Rating().render();
+
+		var elements = rating.element.querySelectorAll('a');
+		dom.triggerEvent(elements[4], 'click');
+
+		setTimeout(function(){
+			assert.strictEqual(rating.rate, 5);
+			done();
+		}, 25);
+	});
+
+	it('should highlight items until mouseover event target element', function(done) {
+		rating = new Rating().render();
+
+		var elements = rating.element.querySelectorAll('a');
+		dom.triggerEvent(elements[4], 'mouseover');
+
+		setTimeout(function(){
+			elements = rating.element.querySelectorAll('a');
+			assert.strictEqual(elements[0].className, rating.cssClasses.on);
+			assert.strictEqual(elements[1].className, rating.cssClasses.on);
+			assert.strictEqual(elements[2].className, rating.cssClasses.on);
+			assert.strictEqual(elements[3].className, rating.cssClasses.on);
+			assert.strictEqual(elements[4].className, rating.cssClasses.on);
+			done();
+		}, 25);
+	});
+
+	it('should highlight the element according with current rate after mouseout event', function(done) {
+		rating = new Rating().render();
+
+		var elements = rating.element.querySelectorAll('a');
+		dom.triggerEvent(elements[4], 'click');
+
+		setTimeout(function(){
+			dom.triggerEvent(elements[4], 'mouseout');
+
+			setTimeout(function(){
+				elements = rating.element.querySelectorAll('a');
+				assert.strictEqual(elements[0].className, rating.cssClasses.on);
+				assert.strictEqual(elements[1].className, rating.cssClasses.on);
+				assert.strictEqual(elements[2].className, rating.cssClasses.on);
+				assert.strictEqual(elements[3].className, rating.cssClasses.on);
+				assert.strictEqual(elements[4].className, rating.cssClasses.on);
+				done();
+			}, 25);
+		}, 25);
+	});
+
+	it('should clear rate if the user click twice in the same element', function(done) {
+		rating = new Rating().render();
+
+		var elements = rating.element.querySelectorAll('a');
+		dom.triggerEvent(elements[4], 'click');
+
+		setTimeout(function(){
```

Comment:
These `setTimeout` calls are most likely being used here because of Metal's asynchronous rendering, right?
Using `setTimeout` is not very safe though, as you can't be so sure about if 25ms is enough for the rendering to be done. 
What I've been using on my tests is an event that the component fired called `stateSynced`, which means that the state changes have already synced the component. Try it out on your tests too :)